plugins/woocommerce-blocks/tests/e2e/tests/price-filter/price-filter.block_theme.spec.ts

```diff
@@ -0,0 +1,409 @@
+
+						return containsPriceRange;
+					}
+					return false;
+				},
+				{ timeout: 3000 }
+			)
+			.catch( () => {
+				// Do nothing. This is only to ensure the products are loaded.
+				// There are multiple requests until the products are fully
+				// loaded. We need to ensure the page is ready to be interacted
+				// with, hence the extra check. Ideally, this should be signaled
+				// by the UI (e.g., by a loading spinner), but we don't have
+				// that yet.
+			} );
+	} );
+
+	test( 'should show all products', async ( { frontendUtils } ) => {
+		const allProductsBlock = await frontendUtils.getBlockByName(
+			'woocommerce/all-products'
+		);
+
+		const img = allProductsBlock.locator( 'img' ).first();
+
+		await expect( img ).not.toHaveAttribute(
+			'src',
+			blockData.placeholderUrl
+		);
+
+		const products = allProductsBlock.getByRole( 'listitem' );
+
+		await expect( products ).toHaveCount( 9 );
+	} );
+
+	test( 'should show only products that match the filter', async ( {
+		page,
+		frontendUtils,
+	} ) => {
+		// The price filter input is initially enabled, but it becomes disabled
+		// for the time it takes to fetch the data. To avoid setting the filter
+		// value before the input is properly initialized, we wait for the input
+		// to be disabled first. This is a safeguard to avoid flakiness which
+		// should be addressed in the code, but All Products block will be
+		// deprecated in the future, so we are not going to optimize it.
+		await page
+			.getByRole( 'textbox', {
+				name: 'Filter products by maximum price',
+				disabled: true,
+			} )
+			.waitFor( { timeout: 3000 } )
+			.catch( () => {
+				// Do not throw in case Playwright doesn't make it in time for the
+				// initial (pre-request) render.
+			} );
+
+		const maxPriceInput = page.getByRole( 'textbox', {
+			name: 'Filter products by maximum price',
+		} );
+		// await page.pause();
+
+		await maxPriceInput.dblclick();
+		await maxPriceInput.fill( '$5' );
+		await maxPriceInput.press( 'Tab' );
+
+		const allProductsBlock = await frontendUtils.getBlockByName(
+			'woocommerce/all-products'
+		);
+
+		const img = allProductsBlock.locator( 'img' ).first();
+		await expect( img ).not.toHaveAttribute(
+			'src',
+			blockData.placeholderUrl
+		);
+
+		const allProducts = allProductsBlock.getByRole( 'listitem' );
+
+		await expect( allProducts ).toHaveCount( 1 );
+		expect( page.url() ).toContain(
+			blockData.urlSearchParamWhenFilterIsApplied
+		);
+	} );
+} );
+test.describe( `${ blockData.name } Block - with PHP classic template`, () => {
+	test.beforeEach( async ( { admin, page, editor } ) => {
+		await cli(
+			'npm run wp-env run tests-cli -- wp option update wc_blocks_use_blockified_product_grid_block_as_template false'
+		);
+
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//archive-product',
+			postType: 'wp_template',
+			canvas: 'edit',
+		} );
+
+		await editor.canvas.locator( 'body' ).click();
```

Comment:
We're already loading with `canvas: edit` so we should be good removing this line.